Library/Homebrew/cmd/reinstall.rb

```diff
@@ -152,5 +153,7 @@ def reinstall
     end
 
     Homebrew.messages.display_messages(display_times: args.display_times?)
+
+    Autoremove.remove_unused_formulae if Homebrew::EnvConfig.autoremove?
```

Comment:
I can add `autoremove` to the `cleanup.rb` namespace but I don't think it should be called by `Cleanup.install_formula_clean!` because that means that we would be running `autoremove` multiple times which is functionally unnecessary.

We could refactor `autoremove` so that it only acts on a formula and it's dependents but I'm not sure if that behavior would be confusing. There is the possibility that unused formula would not get removed because they are not part of a recent install/uninstall/upgrade.

Reply:
`Cleanup.install_formula_clean!` should be the entry point but then it should only be triggered on `periodic_clean!` and then run after all the `cleanup_formula` loops have run but before `cleanup_cache` is run. Make sense?

Reply:
Okay, I think I see what you're saying now. Have a look at the most recent changes and tell me if we're on the same page.